python/lsst/db/db.py

```diff
@@ -54,547 +41,84 @@
+                        (The same as read_default_file in mysql **kwargs)
 
-        If <dbName> is not set, the current database name will be used. Raise
-        exception if the table does not exist and the mustExist flag is not set to
-        False.
-        """
-        dbNameStr = "`%s`." % dbName if dbName is not None else ""
-        try:
-            self.execCommand0("DROP TABLE %s`%s`" % (dbNameStr, tableName))
-        except DbException as e:
-            if e.errCode() == DbException.TB_DOES_NOT_EXIST and not mustExist:
-                self._log.debug("dropTable failed, mustExist is False")
-            else:
-                raise
+        @param user     User to connect as
 
-    def listTables(self, dbName):
-        """
-        Return list of tables in a given database.
-        """
-        cmd = "SELECT TABLE_NAME FROM information_schema.TABLES "
-        cmd += "WHERE TABLE_SCHEMA='%s'" % dbName
-        rows = self.execCommandN(cmd)
-        return [x[0] for x in rows]
+        @param passwd   Password to use
 
-    def isView(self, tableName, dbName=None):
-        """
-        Return True if the table <tableName> is a view, False otherwise.
+        @param host     Host to connect
 
-        @param tableName  Table name.
-        @param dbName     Database name.
+        @param port     TCP/IP port to connect to
 
-        @return boolean   True if the table is a view. False otherwise.
+        @param db       Database to use
 
-        If <dbName> is not set, the current database name will be used.
-        """
-        dbNameStr = "'%s'" % dbName if dbName is not None else "DATABASE()"
-        rows = self.execCommandN("SELECT table_type FROM information_schema.tables "
-               "WHERE table_schema=%s AND table_name='%s'" % (dbNameStr, tableName))
-        return len(rows) == 1 and rows[0][0] == 'VIEW'
+        @param query    Query parameter, as defined by SQLAlchemy
 
-    def getTableContent(self, tableName, dbName=None):
-        """
-        Get contents of the table <tableName>.
+        Initializes SQLAlchemy. There are two ways:
+        1. If defFile is not null, it will be used, and values for: user, passwd,
+           host, port and db will be set and passed to SQLAlchemy.
+        2. If defFile is null, the values for user/passwd/host/port/db/query are
+           simply passed to SQLAlchemy.
 
-        @param tableName  Table name.
-        @param dbName     Database name.
+        Raises DbException if defFile does not exist, or it is incomplete.
 
-        @return string    Contents of the table.
+        Hints:
+        To specify socket, use query={"unix_socket": "/the/socket.file"}
         """
-        dbNameStr = "`%s`." % dbName if dbName is not None else ""
-        ret = self.execCommandN("SELECT * FROM %s`%s`" % (dbNameStr, tableName))
-        s = StringIO.StringIO()
-        s.write(tableName)
-        if len(ret) == 0:
-            s.write(" is empty.\n")
+        if defFile:
```

Comment:
I think the second use case would be useful if it does the translation from the MySQL default param names => SQLAlchemy param names. In that case, parse_config_file would return the raw config file, and engine_from_args would accept those and do the translation (i.e. translate unix_socket, user=>username, etc...) That way if somebody used a separate mechanism for importing configuration from a MySQL defaults file, they wouldn't need to perform that same translation themselves.